src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -65,6 +69,10 @@ namespace NuGetVSExtension
         NuGetConsole.GuidList.GuidPackageManagerConsoleFontAndColorCategoryString,
         "{" + GuidList.guidNuGetPkgString + "}")]
     [Guid(GuidList.guidNuGetPkgString)]
+
+    [ProvideBrokeredService(NuGetBrokeredServices.SourceRepositoryProviderServiceName,
+                            NuGetBrokeredServices.SourceRepositoryProviderServiceVersion,
+                            Audience = ServiceAudience.RemoteExclusiveClient | ServiceAudience.Local)]
```

Comment:
An idea, which might be an overkill right now. 
I think it'd be nice for a guest to be able to view the sources, but not to be able to edit them.

Reply:
This suggestion has a split: there are owner guests and non-owner guests. You can say `AllClientsIncludingGuests` to enable Live Share *owner* guests to access this service.
You can further add the secondary property on this attribute to allow *non-owner* guests to also access the service, which you would do only after filing a thread model, adding the necessary authorization service calls, etc. I think offering readonly access to non-owner guests is a fine idea.